cli/src/main/java/org/eclipse/hono/cli/client/ApplicationClient.java

```diff
@@ -0,0 +1,321 @@
+import org.eclipse.hono.client.ApplicationClientFactory;
+import org.eclipse.hono.client.HonoConnection;
+import org.eclipse.hono.config.ClientConfigProperties;
+import org.eclipse.hono.util.Strings;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.shell.standard.ShellCommandGroup;
+import org.springframework.shell.standard.ShellComponent;
+import org.springframework.shell.standard.ShellMethod;
+import org.springframework.shell.standard.ShellOption;
+import org.springframework.stereotype.Component;
+
+import io.vertx.core.Vertx;
+
+/**
+ * Being a Component will be injected by spring.
+ */
+@Component
+public class ApplicationClient extends AbstractCliClient {
+    /**
+     * Spring Shell output helper.
+     * <p>
+     * it can be passed to the method class instead to use the log (it's more stylish).
+     */
+    @Autowired
+    public ShellHelper shellHelper;
+    /**
+     * Spring Shell inputReader.
+     */
+    @Autowired
+    public InputReader inputReader;
+    /**
+     * Class that contains all parameters for the connection to Hono and for the execution of the commands.
+     * Handling it a part allow to clone it and maintain a default config during the whole execution.
+     */
+    protected ClientConfig clientConfig;
+    /**
+     * All parameters that can be setted with the spring configuration at startup.
+     * There will be injected with values and the stored in the configuration class, to allow the overriding of the parameters.
+     */
+    @Value(value = "${tenant.id:DEFAULT_TENANT}")
+    private String tenantId;
+    @Value(value = "${device.id}")
+    private String deviceId;
+    @Value(value = "${message.type:all}")
+    private String messageType;
+    @Value(value = "${connection.retryInterval:1000}")
+    private int connectionRetryInterval;
+    @Value(value = "${command.timeoutInSeconds:60}")
+    private int commandTimeoutInSeconds;
+    @Value(value = "${message.address:127.0.0.1}")//The address to set on the message being sent.
+    private String messageAddress;
+    @Value(value = "${message.payload:}")//The payload to send.
+    private String messagePayload;
+    /**
+     * To stop the executions of internal commands.
+     */
+    private CountDownLatch latch;
+
+    /**
+     * Sets the vert.x instance.
+     *
+     * @param vertx The vert.x instance.
+     * @throws NullPointerException if vert.x is {@code null}.
+     */
+    @Autowired
+    public final void setVertx(final Vertx vertx) {
+        this.vertx = Objects.requireNonNull(vertx);
+        this.ctx = vertx.getOrCreateContext();
+    }
+    /**
+     * Sets the default client configuration with the parameters got at startup.
+     *
+     * @param honoClientConfig The Client Configuration class instance.
+     */
+    @Autowired
+    public final void setClientConfigProperties(final ClientConfigProperties honoClientConfig) {
```

Comment:
what do we need this for? Aren't all necessary parameters being provided when entering a command in the shell?